app/models.py

```diff
@@ -482,6 +480,31 @@ def personalisation(self, personalisation):
         if personalisation:
             self._personalisation = encryption.encrypt(personalisation)
 
+    @staticmethod
+    def from_api_request(
```

Comment:
The Pythonic way to do this is with a class method, eg
```
@classmethod
def from_api_request(cls, args):
    return cls(args)
```
Means that if you ever subclass `Notification` then the method will always call its constructor, not the parent class.